core/src/main/java/org/mule/config/bootstrap/AbstractRegistryBootstrap.java

```diff
@@ -253,6 +253,10 @@ private void registerObject(String key, String value) throws Exception
         {
             throwExceptionIfNotOptional(optional, cnfe, "Ignoring optional object: " + className);
         }
+        catch (NoSuchMethodException nsme)
```

Comment:
just somehting that i found during testing... That exception could also happen with abstract classes or classes without a default constructor... not directly related but worth accounting for